spec/lib/extensions/ar_to_model_hash_spec.rb

```diff
@@ -30,6 +30,7 @@
 
     let(:fixed_options)       { test_vm_class.send(:to_model_hash_options_fixup, @test_to_model_hash_options) }
     let(:mocked_preloader)    { double }
+    let(:rails7)              { !ActiveRecord::Associations::Preloader.instance_methods.include?(:preload) }
```

Comment:
Note, I'll drop this conditional unless we can figure out how to get this to be backward compatible with 6.1 as the query counts are different.

Reply:
drop this, rails 7 all the way